### ISSUE TITLE
Audacity creates an extra Envelope Point when copying a selection to the same track.

### ISSUE THREAD

**Issue by JohnColket** [audacity/audacity#4076]
### Bug description

Regression over 3.1.3.  When copying a clip from one part of a track to another part of the same track, Audacity 3.2.2 seems to add three (3) Envelope Points.  3.1.3 adds two (2) points. This "bug" was identified during testing trying to find out why extra envelope points seem to be appearing randomly in forum users projects, and how "millions" of envelope points could occur in a single project. See #407

In addition to the "extra" envelope point, the copied clip seems to be rendered incorrectly.  Rendering in 3.1.3 seems to be correct.

### Steps to reproduce

1) Generate 20 second tone
2) Select Envelope Tool (F2)
3) Add an envelope point at the 2, 3, and 4 second marks.
4) Select Selection Tool (F1)
5) Swipe (select) audio from 1 to 5 second mark.
6) Ctrl+C = Copy
7) Position cursor to 10 second mark
8) Ctrl+V = Paste

### Expected behavior

Copied Waveform matches original as in 3.1.3
Extra > Scriptables II > Get Info > Envelopes shows 8 = 3 + 2 + 3 envelope points.
![Tracks313](https://user-images.githubusercontent.com/76828213/208155254-9eeeea7d-2387-4cd1-b99a-e960e4cd114a.png)

### Actual behavior

Copied Waveform does not match original
Extra > Scriptables II > Get Info > Envelopes shows 9 = 3 + 3 + 3 envelope points.
![Tracks322](https://user-images.githubusercontent.com/76828213/208155275-8b7ed3da-3c74-469d-9828-0e18ed2a0131.png)

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

### Additional context

See also #4035 and #4070 which may be related.

**Comment by petersampsonaudacity** [comment]
Testing on W10 with 3.2.2 I cannot reproduce this following the STRs

![image](https://user-images.githubusercontent.com/50205984/208299246-b379c86a-d69b-4e0e-982f-a8c651174b04.png)

**Comment by JohnColket** [comment]
> Testing on W10 with 3.2.2 (or 3.2.3 beta 0r 3.3.0 alpha) I cannot reproduce this following the STRs

@petersampsonaudacity and Extra > Scriptables II > Get Info > Envelopes shows ? envelope points?

**Comment by JohnColket** [comment]
Yes, @petersampsonaudacity,  the shape is not always distorted.  It is only distorted for certain envelope point selections.  The extra envelope point always seems to be present.  You should be able to reproduce the irregular shape if you drop the first envelope point a just a little bit more.

![Tracks025](https://user-images.githubusercontent.com/76828213/208316984-6b54c9e5-c84a-4465-83e8-9b6da17dd58b.png)

**Comment by petersampsonaudacity** [comment]
@JohnColket   testing on W10 with latest 3.2.3 Beta:  audacity-win-3.2.3-beta-20221219+0b46906-x64-msvc2022

And using your suggested dropped envelope point I still don't get the extra envelop points or distortion
![image](https://user-images.githubusercontent.com/50205984/208427164-018ada6b-43a9-47d9-8551-1a1741ce5d4f.png)

**Comment by JohnColket** [comment]
Thank you @petersampsonaudacity .  I didn't realize the 3.2.3 beta was out.  The problem is still present (for me) with 3.2.3 beta (and I still get the extra envelope point). Did you do ever do Extras > Scriptables II > Get Info > Envelope  to count the envelope points?

**Comment by petersampsonaudacity** [comment]
> I didn't realize the 3.2.3 beta was out. The problem is still present (for me) with 3.2.3 beta (and I still get the extra envelope point). 

Its in GitHub builds but not on general testing release yet is my understanding - and there's a 3,3.0 alpha in among the GitHub builds (it's the one that that goes under the moniker "master"

>Did you do ever do Extras > Scriptables II > Get Info > Envelope to count the envelope points?

Phew, no I never touch those Darke Arts Blacke Magicke spelles...

**Comment by JohnColket** [comment]
> Phew, no I never touch those Darke Arts Blacke Magicke spelles...

That is wherein the problem lies...

**Comment by dozzzzer** [comment]
I was able to reproduce this on the 2nd attempt, it created an extra env point:
<img width="600" alt="image" src="https://user-images.githubusercontent.com/70063970/221636574-f62e6f56-b2c8-4be1-921f-e91769371e48.png">

OS: Win 11
Version: 3.2.4
